include/ck_tile/ops/gemm/kernel/gemm_kernel.hpp

```diff
@@ -303,21 +304,63 @@ struct GemmKernel
         const auto& b_tensor_view = [&]() {
             if constexpr(std::is_same_v<BLayout, tensor_layout::gemm::RowMajor>)
             {
-                return make_naive_tensor_view<address_space_enum::global>(
-                    b_ptr,
-                    make_tuple(splitk_batch_offset.splitted_k, kargs.N),
-                    make_tuple(kargs.stride_B, 1),
-                    number<GemmPipeline::GetVectorSizeB()>{},
-                    number<1>{});
+                if constexpr(TilePartitioner::BlockGemmShape::PermuteB)
+                {
+                    const index_t K1 = TilePartitioner::BlockGemmShape::kK;
+                    const index_t K0 =
+                        splitk_batch_offset.splitted_k / TilePartitioner::BlockGemmShape::kK;
+                    const auto b_k0_n_k1_desc =
+                        make_naive_tensor_descriptor(make_tuple(K0, kargs.N, K1),
+                                                     make_tuple(kargs.N * K1, K1, I1),
+                                                     number<GemmPipeline::GetVectorSizeB()>{},
+                                                     number<1>{});
+                    const auto b_n_k_desc = transform_tensor_descriptor(
```

Comment:
I wonder whether this 3D tensor descriptor would work well with 2DStaticTileEncodingPattern?

Reply:
It is descriptor but 2DStaticTileEncodingPattern implements encodings